subprojects/maven/src/integTest/groovy/org/gradle/api/publish/maven/MavenPublishMultiProjectIntegTest.groovy

```diff
@@ -466,7 +473,14 @@ project(":library") {
         and:
         resolveArtifacts(platformModule) { expectFiles 'foo-1.0.jar' }
         resolveArtifacts(libraryModule) {
-            expectFiles 'bar-1.1.jar', 'foo-1.0.jar', 'library-1.0.jar'
+            withModuleMetadata {
+                expectFiles 'bar-1.1.jar', 'foo-1.0.jar', 'library-1.0.jar'
+            }
+            withoutModuleMetadata {
+                // This is caused by the dependency on the platform appearing as a dependencyManagement entry with scope=import, type=pom
+                // and thus its dependencies are ignored.
+                expectFiles 'bar-1.1.jar', 'library-1.0.jar'
```

Comment:
This assertion and its comment are probably worth a longer discussion. It illustrates how feature combination ends up with an impact on dependency graph resolution based on the metadata format.